source/common/router/router.cc

```diff
@@ -1619,6 +1622,17 @@ bool Filter::convertRequestHeadersForInternalRedirect(Http::RequestHeaderMap& do
   return true;
 }
 
+void Filter::runRetryOptionsPredicates(UpstreamRequest& retriable_request) {
+  for (const auto& options_predicate : route_entry_->retryPolicy().retryOptionsPredicates()) {
+    const Upstream::RetryOptionsPredicate::UpdateOptionsParameters parameters{
```

Comment:
The docs say that the parameter will be the previous attempt's options, but I think by rebuilding the options on each loop iteration, if an earlier predicate updates the options, that update will be passed to the later predicates.